### PR TITLE
Update @embroider/macros examples in the readme

### PR DESCRIPTION
**Pull request by thoov**
Not the full list of macros but a first pass on the big ones

### CODE REVIEW

File: packages/macros/README.md [hunk 1]
```diff
@@ -8,61 +8,179 @@ Traditionally, Ember addons have a lot of power to run arbitrary code during the
...
-         setConfig: {
-           'some-dependency': {
-             // config for some-dependency
-           },
-         },
-       },
-     },
-   };
-   ```
+- [ember-exam](https://github.com/ember-cli/ember-exam)
```
Comment (simonihmig) [inline packages/macros/README.md:185]: I looked at this to find out what they are using macros for, but couldn't find any usage!? 🤔

File: packages/macros/README.md [hunk 2]
```diff
@@ -8,61 +8,179 @@ Traditionally, Ember addons have a lot of power to run arbitrary code during the
...
+- [ember-bootstrap](https://github.com/kaliber5/ember-bootstrap)
```
Comment (simonihmig) [inline packages/macros/README.md:186]: If you want to add another example, I use macros in tests only for https://github.com/kaliber5/ember-stargate, basically as a replacement for `ember-compatibility-helpers`

File: packages/macros/README.md [hunk 3]
```diff
@@ -8,61 +8,179 @@ Traditionally, Ember addons have a lot of power to run arbitrary code during the
...
+}
+```
+
+```hbs
+{{macroDependencySatisfies 'qunit' '^2.8.0'}}
+```
+
+### getOwnConfig, getConfig, and getGlobalConfig
+
+A common pattern is to have a set of configuration properties that you define (or a consumer defines for you) which you base certain build time conditions around. This is achieved via the `getOwnConfig`, `getConfig`, and `getGlobalConfig` macros (depending on which config you want to read).
```
Comment (simonihmig) [inline packages/macros/README.md:148]: Should we explain those in a bit more detail? Like `getGlobalConfig` was new to me, and not mentioned in the RFC, so it's not obvious what a global config is.

**Review by simonihmig — COMMENTED** [review]
👍

File: packages/macros/README.md (hunk 1 above)
Comment (thoov) [reply packages/macros/README.md:185]: Its coming with my open PR: https://github.com/ember-cli/ember-exam/pull/599

File: packages/macros/README.md (hunk 2 above)
Comment (thoov) [reply packages/macros/README.md:186]: 👍  will add

File: packages/macros/README.md [hunk 4]
```diff
@@ -8,61 +8,180 @@ Traditionally, Ember addons have a lot of power to run arbitrary code during the
 
 The [Embroider package spec](../../SPEC.md) proposes fixing this by making Ember addons much more static. But they will still need the ability to change themselves in certain ways at app compilation time. Hence this package.
 
-This package works in both Embroider builds and normal ember-cli builds, so that addon authors can switch to this newer pattern without disruption.
+This package works in both Embroider and Classical builds, so that addon authors can switch to this newer pattern without disruption.
```
Comment (acorncom) [inline packages/macros/README.md:11]: Might read more consistently across the ecosystem to call these Classic builds?

File: packages/macros/README.md [hunk 5]
```diff
@@ -8,61 +8,180 @@ Traditionally, Ember addons have a lot of power to run arbitrary code during the
...
+    },
+  },
+};
+```
 
 ## The Macros
 
+### macroCondition
+
+The `macroCondition` macro allows branch level code isolation (and deletion in the case of production builds). Generally macroConditions are viewed as a foundation macro and are combined with others marcos (detailed below) to create more complex scenarios. `macroCondition` takes a single argument which must be statically known or another macro which will compile down to a static value.
```
Comment (acorncom) [inline packages/macros/README.md:64]: combined with other **macros**

**Review by rwjblue — APPROVED** [review]
I'd have loved these docs when I recently tried to use `@embroider/macros` instead of `ember-compatibility-helpers`...

🙏

**Pull request merged**